### PR TITLE
Integrate debug test framework

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will integrate debug test framework to `ballerina-distribution`. By integrating the debug framework we can now write debug test cases with standard libraries as well.

Fixes: https://github.com/ballerina-platform/ballerina-lang/issues/26813

### CODE REVIEW

**Review by NipunaRanasinghe — COMMENTED** [review]
@praveennadarajah Shall we add the debugger integration tests as a part of the `devtools-integration-tests`? WDYT?

**Comment by praveennadarajah** [comment]
> @praveennadarajah Shall we add the debugger integration tests as a part of the `devtools-integration-tests`? WDYT?

@NipunaRanasinghe IMO, having a separate module for `debbuger-integration-tests` is better. This will help us to maintain the same structure with [ballerina-lang](https://github.com/ballerina-platform/ballerina-lang/tree/master/tests/jballerina-debugger-integration-test). Also according to the implementation we should not change the structure of `src/test/resources` folder. Also we should not change the names of the directory inside `src/test/resources`. If we do a changes here, we have to adjust accordingly in `ballerina-lang`. Therefore having a separate module is better. WDYT?

**Comment by NipunaRanasinghe** [comment]
> > @praveennadarajah Shall we add the debugger integration tests as a part of the `devtools-integration-tests`? WDYT?
> 
> @NipunaRanasinghe IMO, having a separate module for `debbuger-integration-tests` is better. This will help us to maintain the same structure with [ballerina-lang](https://github.com/ballerina-platform/ballerina-lang/tree/master/tests/jballerina-debugger-integration-test). Also according to the implementation we should not change the structure of `src/test/resources` folder. Also we should not change the names of the directory inside `src/test/resources`. If we do a changes here, we have to adjust accordingly in `ballerina-lang`. Therefore having a separate module is better. WDYT?

@praveennadarajah As per the discussion we had, lets maintain these tests under the `dev-tools-integration-tests`.

File: build.gradle [hunk 1]
```diff
@@ -30,6 +30,7 @@ ext {
     dockerVersion = project.ballerinaVersion
     dataMapperVersion = project.ballerinaVersion
     c2cVersion = project.ballerinaVersion
+    ballerinaNextSnapshotVersion = project.ballerinaNextSnapshotVersion
```
Comment (NipunaRanasinghe) [inline build.gradle:33]: Lets use ` project.ballerinaVersion` instead

File: gradle.properties [hunk 2]
```diff
@@ -1,6 +1,7 @@
 org.gradle.caching=true
 group=org.ballerinalang
 version=swan-lake-preview8-SNAPSHOT
+ballerinaNextSnapshotVersion=2.0.0-Preview8-SNAPSHOT
```
Comment (NipunaRanasinghe) [inline gradle.properties:4]: lets remove this constant and use `ballerinaVersion`

File: debugger-integration-tests/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import org.ballerinalang.test.context.BallerinaTestException;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+
+import java.io.IOException;
+
+/**
+ * Parent test class for all debug integration test cases.
+ */
+public class DebugAdapterBaseTestCase {
```
Comment (NipunaRanasinghe) [inline debugger-integration-tests/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java:30]:
```suggestion
public class BaseTestCase {
```

**Comment by praveennadarajah** [comment]
> > > @praveennadarajah Shall we add the debugger integration tests as a part of the `devtools-integration-tests`? WDYT?
> > 
> > 
> > @NipunaRanasinghe IMO, having a separate module for `debbuger-integration-tests` is better. This will help us to maintain the same structure with [ballerina-lang](https://github.com/ballerina-platform/ballerina-lang/tree/master/tests/jballerina-debugger-integration-test). Also according to the implementation we should not change the structure of `src/test/resources` folder. Also we should not change the names of the directory inside `src/test/resources`. If we do a changes here, we have to adjust accordingly in `ballerina-lang`. Therefore having a separate module is better. WDYT?
> 
> @praveennadarajah As per the discussion we had, lets maintain these tests under the `dev-tools-integration-tests`.

@NipunaRanasinghe Moved the tests to `devtools-integration-tests`

File: build.gradle (hunk 1 above)
Comment (praveennadarajah) [reply build.gradle]: Done.

File: debugger-integration-tests/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java (hunk 3 above)
Comment (praveennadarajah) [reply debugger-integration-tests/src/test/java/org/ballerinalang/debugger/test/DebugAdapterBaseTestCase.java]: Done.

File: gradle.properties (hunk 2 above)
Comment (praveennadarajah) [reply gradle.properties]: Done.

File: devtools-integration-tests/src/test/java/org/ballerina/devtools/debug/ModuleBreakpointTest.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.ballerinalang.debugger.test.utils.DebugTestRunner.DebugResumeKind;
+
+/**
+ * Test class for ballerina breakpoints related test scenarios.
+ */
+public class ModuleBreakpointTest extends BaseTestCase {
```
Comment (NipunaRanasinghe) [inline devtools-integration-tests/src/test/java/org/ballerina/devtools/debug/ModuleBreakpointTest.java:36]: Is there any specific reason for moving the `ModuleBreakpointTest` into the distribution?

Comment (praveennadarajah) [reply devtools-integration-tests/src/test/java/org/ballerina/devtools/debug/ModuleBreakpointTest.java:36]: This `ModuleBreakpointTest` is to ensure that debug engage is working fine with standard libraries.

Comment (NipunaRanasinghe) [reply devtools-integration-tests/src/test/java/org/ballerina/devtools/debug/ModuleBreakpointTest.java:36]: Ah I see.. +1

**Review by NipunaRanasinghe — APPROVED** [review]
Looks good! Lets create an issue to track the disabled test cases.

**Comment by praveennadarajah** [comment]
> Looks good! Lets create an issue to track the disabled test cases.

Issue created to enable ServiceDebugTest #1016

File: build.gradle [hunk 5]
```diff
@@ -479,6 +486,21 @@ subprojects {
...
+        /* Azure Functions extension */
+        exten "org.ballerinax.azurefunctions:azurefunctions-extension:${azFunctionsVersion}"
+        azFunctionsBalo "org.ballerinax.azurefunctions:azurefunctions-extension-balo:${azFunctionsVersion}"
+        azFunctionsExamples "org.ballerinax.azurefunctions:azurefunctions-extension-examples:${azFunctionsVersion}@zip"
+
+        /* Docker extension */
+        exten "org.ballerinax.docker:docker-extension:${dockerVersion}@jar"
+        dockerAnnotations "org.ballerinax.docker:docker-extension-annotations:${dockerVersion}"
+        dockerExamples "org.ballerinax.docker:docker-extension-examples:${dockerVersion}@zip"
+
```
Comment (niveathika) [inline build.gradle:503]: @praveennadarajah Seems these are added due to issues in git merge. Review the PR and revert any unnecessary changes

Comment (praveennadarajah) [reply build.gradle:503]: @niveathika Removed unnecessary changes via [#1068](https://github.com/ballerina-platform/ballerina-distribution/pull/1068)
